style/layer/highway_shield.js

```diff
@@ -1,49 +1,69 @@
 "use strict";
 
-export const interstate = {
-  id: "highway-shield-us-interstate",
+import * as Util from "../js/util.js";
+
+function routeConcurrency(num) {
+  return [
+    "case",
+    ["!=", ["get", "route_" + num], null],
+    ["image", ["concat", "shield_", ["get", "route_" + num]]],
+    ["literal", ""],
+  ];
+}
+
+function routeSpacer(num) {
+  return [
+    "case",
+    ["!=", ["get", "route_" + num], null],
+    ["image", "spacer"],
```

Comment:
Setting [`text-letter-spacing`](https://maplibre.org/maplibre-gl-js-docs/style-spec/layers/#layout-symbol-text-letter-spacing) on the layer would likely spread out the images within the `format` expression without the need for a spacer image.

Reply:
Thanks!  Implemented in 3331b82